rancher2-deployment/Dockerfile

```diff
@@ -1,23 +1,57 @@
 FROM rancher/cli2 AS release
-ARG DEPLOYMENT_TEMPLATE_FILE="deployment.tmpl.yml"
-ENV DEPLOYMENT_TEMPLATE_FILE=${DEPLOYMENT_TEMPLATE_FILE}
 
-# Mandatory rancher context:
+# This image's main purpose is to run deployments to a Rancher2 cluster. It
+# leverages the rancher2-cli, kubectl and gomplate to accomplish that.
+
+
+########################################################################
+# Mandatory variables
+
+# Define the name of k8s configuration.
+ENV DEPLOYMENT_TEMPLATE_FILE="deployment.tmpl.yml"
+
+# Sets the rancher context containing of <cluster-id>:<project-id>.
 ENV RANCHER_CONTEXT=""
 
-# Mandatory project name:
-ENV PROJECT=""
+# Sets the project name. Must be the name of the Rancher2 project. The value
+# is used together with the value of ENVIRONMENT to construct the name of the
+# K8s namespace. Can be overriden by PROJECT.
```

Comment:
where is PROJECT?

Reply:
The `PROJECT` env var must be set in the specific project repo. Usually this is a constant value in the `Makefile` in the root of a project repo.

Reply:
I changed some to make this less confusing.